scripts/create-github-release.js

```diff
@@ -0,0 +1,184 @@
+    owner: GH_OWNER,
+    repo: GH_REPO,
+    ref: `tags/${tagName}`,
+  });
+  if (refData.status !== 200) {
+    console.error('refData:');
+    console.error(refData);
+    throw new Error(
+      'Something went wrong when getting the tag SHA using tag name',
+    );
+  }
+  const tagSha = refData.data.object.sha;
+  console.log(`SHA for the tag ${TAG_NAME} is ${tagSha}`);
+
+  // Get the commit SHA using the tag SHA
+  const tagData = await octokit.git.getTag({
+    owner: GH_REPO,
+    repo: GH_REPO,
+    tag_sha: tagSha,
+  });
+  if (tagData.status !== 200) {
+    console.error('tagData:');
+    console.error(tagData);
+    throw new Error(
+      'Something went wrong when getting the commit SHA using tag SHA',
+    );
+  }
+  const commitSha = tagData.data.object.sha;
+  console.log(
+    `The commit for the tag is https://github.com/backstage/backstage/commit/${commitSha}`,
+  );
+
+  // Get the commit message using the commit SHA
+  const commitData = await octokit.git.getCommit({
+    owner: GH_OWNER,
+    repo: GH_REPO,
+    commit_sha: commitSha,
+  });
+  if (commitData.status !== 200) {
+    console.error('commitData:');
+    console.error(commitData);
+    throw new Error(
+      'Something went wrong when getting the commit message using commit SHA',
+    );
+  }
+
+  // Example Commit Message
+  // Merge pull request #3555 from backstage/changeset-release/master Version Packages
+  return commitData.data.message;
+}
+
+// There is a PR number in our expected commit message. Get the description of that PR.
+async function getPrDescriptionFromCommitMessage(commitMessage) {
+  // It should exactly match the pattern of changeset commit message, or else will abort.
+  const expectedMessage = RegExp(EXPECTED_COMMIT_MESSAGE);
+  if (!expectedMessage.test(commitMessage)) {
+    throw new Error(
+      `Expected regex did not match commit message: ${commitMessage}`,
+    );
+  }
+
+  // Get the PR description from the commit message
+  const prNumber = commitMessage.match(expectedMessage).groups.prNumber;
+  console.log(
+    `Identified the changeset Pull request - https://github.com/backstage/backstage/pull/${prNumber}`,
+  );
+
+  const prData = await octokit.pulls.get({
+    owner: GH_OWNER,
+    repo: GH_REPO,
+    pull_number: prNumber,
+  });
+
+  return prData.data.body;
+}
+
+// Use the PR description to prepare for the release description
+async function prepareReleaseDescription(prDescription) {
+  // TODO: Refine prDescription to remove the lines containing "Update Dependencies"
+  // Remove everything in the beginning until changelogs.
+  return prDescription.split('\n').slice(3).join('\n');
+}
+
+// Create Release on GitHub.
+async function createRelease(releaseDescription) {
+  // Create the release
+  const releaseResponse = await octokit.repos.createRelease({
+    owner: GH_REPO,
+    repo: GH_REPO,
+    tag_name: TAG_NAME,
+    name: TAG_NAME,
+    body: releaseDescription,
+    draft: false,
```

Comment:
As your next step, one possibility would be to pass your dryrun status into this method here as a `draft: true`.  I believe only repo admins can even _see_ draft releases, which gives you a human way to validate it befre you actually click the "publish button" ?  And if it works for your next release, then you could swing off the dryrun flag in the workflow?

just an idea!

Reply:
Love the idea @adamdmharvey ! 😁